### PR TITLE
Feature/t-step joint

### PR DESCRIPTION
**Pull request by jonashaldemann**
### What type of change is this?

- [ ] Bug fix in a **backwards-compatible** manner.
- [x] New feature in a **backwards-compatible** manner.
- [ ] Breaking change: bug fix or new feature that involve incompatible API changes.
- [ ] Other (e.g. doc update, configuration, etc)

### Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._

- [x] I added a line to the `CHANGELOG.md` file in the `Unreleased` section under the most fitting heading (e.g. `Added`, `Changed`, `Removed`).
- [x] I ran all tests on my computer and it's all green (i.e. `invoke test`).
- [x] I ran lint on my computer and there are no errors (i.e. `invoke lint`).
- [ ] I added new functions/classes and made them available on a second-level import, e.g. `compas_timber.datastructures.Beam`.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [ ] I have added necessary documentation (if appropriate)

### CODE REVIEW

**Comment by jonashaldemann** [comment]
@chenkasirer maybe there is some simplification potential in the code where I added a TODO comment...

**Review by chenkasirer — CHANGES_REQUESTED** [review]
Sorry I took so long. Looks nice! I made a first pass of comments. Let me know if there's anything that's not clear.

File: src/compas_timber/connections/t_step.py [hunk 1]
```diff
@@ -0,0 +1,169 @@
...
+        intersection = intersection_plane_plane(plane1, plane2)
+        rotation_axis = Vector.from_start_end(*intersection)
+        origin = intersection[0]
+        R = Rotation.from_axis_and_angle(rotation_axis, math.radians(90))
+        bisector.transform(R)
+        plane = Plane(origin, bisector)
+        return plane
+
+    # find the Face on cross_beam where main_beam intersects
+    # TODO simplify with Chen!
```
[the remark this answers is not in the document]
Comment (jonashaldemann) [reply src/compas_timber/connections/t_step.py:70]: You are absolutely right. I have no idea how I came up with this :-/

File: src/compas_timber/connections/t_step.py [hunk 2]
```diff
@@ -0,0 +1,169 @@
...
+    def beams(self):
+        return [self.main_beam, self.cross_beam]
+
+    @staticmethod
+    def _bisector_plane(plane1, plane2, angle_factor):
+        bisector = plane1.normal + plane2.normal * angle_factor
+        intersection = intersection_plane_plane(plane1, plane2)
+        rotation_axis = Vector.from_start_end(*intersection)
+        origin = intersection[0]
+        R = Rotation.from_axis_and_angle(rotation_axis, math.radians(90))
```
[the remark this answers is not in the document]
Comment (jonashaldemann) [reply src/compas_timber/connections/t_step.py:64]: You are absolutely right. I have no idea how I came up with this :-/

**Comment by jonashaldemann** [comment]
currently being implemented by @papachap

**Pull request closed without merging**